pkg/orchestration/wiring/kubeingress/plugin.go

```diff
@@ -124,11 +125,9 @@ func buildServiceResource(deploymentName smith_v1.ResourceName, selectorLabels m
 	return serviceResource
 }
 
-// buildIngressResource constructs the Kube / KITT Ingress object
-// with a default rule, plus all aliases rules from dependants internalDNS resources
-func buildIngressResource(serviceName smith_v1.ResourceName, resource *orch_v1.StateResource, context *wiringplugin.WiringContext) (wiringplugin.WiredSmithResource, error) {
+func buildIngressResourceFromSpec(serviceName smith_v1.ResourceName, resourceName voyager.ResourceName, timeout int, context *wiringplugin.WiringContext) (*wiringplugin.WiredSmithResource, error) {
 	var ingressRules []ext_v1b1.IngressRule
-	ingressName := string(resource.Name) + "-" + ingressPostfix
+	ingressName := string(resourceName) + "-" + ingressPostfix
```

Comment:
There is a set of special functions to construct names - `pkg/orchestration/wiring/wiringutil/name.go`